SDWebImage/SDAnimatedImageView.m

```diff
@@ -639,7 +639,7 @@ - (void)updateIsProgressiveWithImage:(UIImage *)image
             } else if (currentData.length > previousData.length) {
                 // If current data is appended by previous data, use `NSDataSearchAnchored`
                 NSRange range = [currentData rangeOfData:previousData options:NSDataSearchAnchored range:NSMakeRange(0, previousData.length)];
-                if (range.location == 0 && range.length == previousData.length) {
+                if (range.location == 0) {
```

Comment:
The reason to change this ? We need to check that the current data contains hole previous data, not a subsequence of previous data. So that `range.length == previousData.length` is necessary.

Reply:
Emm, if `range.location` is valid, that is not `NSNotFound`, `range.length` always equal to `previousData.length`.

Reply:
@zhongwuzw Is that the documented behavior ? This simple check seems not cause any performance cost, and make the code more readable. I think leave it does not seems cause anything and I test it the length is actually match `previousData.length` during stress test

Reply:
I see...So it means actually this API is implemented as this ?

```objectivec
- (NSRange)rangeOfData:(NSData *)dataToFind options:(NSDataSearchOptions)mask range:(NSRange)searchRange {
   NSUInteger loc = [self beginOfData:dataToFind options:mask range:searchRange]; 
   return NSMakeRange(loc, dataToFind.length); // Just a pass the value out
}

- (NSUInteger)beginOfData:(NSData *)dataToFind options:(NSDataSearchOptions)mask range:(NSRange)searchRange {
 //... actual implementation 
}
```

Anyway, I think add this protect is OK. Which does not cost any performance cost and make code readable.

Reply:
Yes, we can assume it works like this, I think the reason why returned `NSRange` here is many `API`s use `NSRange` to do the job, it's just for facility IMO 🤔 .

Yeah, it's trivial, I can revert this change, and in consideration of code readable, I think maybe change `range.location == 0 && range.length == previousData.length` to `range.location != NSNotFound` is more readable? because we call `rangeOfData` and pass `NSDataSearchAnchored`, that is to say, matching `dataToFind` from beginning. So if we get `NSNotFound`, it's meaning not found. @dreampiggy what's your opinion, I'm ok for all these.

Reply:
Emm....If that document's behavior is correc, I think change it into `range.location != NSNotFound` is more suitable. Did you test that if we provide `NSDataSearchAnchored`, the range.location is either `NSNotFound` or `0`, no other value can be returned ? Maybe you can try a demo.

Reply:
@dreampiggy 👌 According to the document, and also under some tests to ensure the correctness, it's ok and works fine.